### ISSUE TITLE
Using `Final` exported from a module gives an error

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#3012]
**Describe the bug**
Given the following files:

mypackage/__init__.py:
```python
```

mypackage/compat.py
```python
import sys


if sys.version_info >= (3, 8):
    from typing import Final as Final
else:
    from typing_extensions import Final as Final
```

mypackage/common.py
```python
from .compat import Final


SOME_CONSTANT: Final = 'asdf'
```

An error is generated on line 4 of `common.py`: `Expression of type "Literal['asdf']" cannot be assigned to declared type "Final"`

**Expected behavior**
No error should be generated and the type of `SOME_CONSTANT` should end up as `Literal['asdf']`.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.219

**Comment by erictraut** [comment]
Yes, this is a known limitation. The meaning of the `Final` symbol needs to be understood in pyright's binder long before types can be analyzed. There's a bunch of special-case code in the binder to detect whether the `Final` symbol is imported from `typing` or `typing_extensions`. In your example, `common.py` imports `Final` from a different location. 

This is a hard limitation, so you'll need to find a workaround.

Here's one possibility:
```python
if TYPE_CHECKING:
    from typing import Final
else:
    from .compat import Final
```
